src/View/ViewableData.php

```diff
@@ -559,15 +581,25 @@ public function obj($fieldName, $arguments = [], $cache = false, $cacheName = nu
             $value = $this->$fieldName;
         }
 
+        // Try to cast object if we have an explicit cast set
+        if (!is_object($value)) {
```

Comment:
Keeping this condition here (mirroring the below) because if a method returns an explicit object (usually `DBField`), the implication is that it shouldn't be re-cast.

We could choose to change that, but if so it should be handled in a separate card with an issue for that purpose.